phys2denoise/workflow.py

```diff
@@ -59,187 +63,252 @@ def save_bash_call(outdir):
-                    regr[f"{key}_sin"] = retroicor_regrs[vslice][:, harm * 2 + 1]
-        elif metric == "retroicor_resp":
-            args = select_input_args(retroicor, kwargs)
-            args["resp"] = True
-            retroicor_regrs = retroicor(physio, **args)
-            for vslice in range(len(args["slice_timings"])):
-                for harm in range(args["n_harm"]):
-                    key = f"rcor-resp_s-{vslice}_hrm-{harm}"
-                    regr[f"{key}_cos"] = retroicor_regrs[vslice][:, harm * 2]
-                    regr[f"{key}_sin"] = retroicor_regrs[vslice][:, harm * 2 + 1]
-        else:
-            args = select_input_args(metric, kwargs)
-            regr[metric.__name__] = metric(physio, **args)
-
-    # #!# Add regressors visualisation
-
-    # Export regressors and sidecar
-    out_filename = os.join(outdir, "derivatives", filename)
-    regr.to_csv(out_filename, sep="\t", index=False, float_format="%.6e")
-    # #!# Add sidecar export
-
-
-def _main(argv=None):
-    options = _get_parser().parse_args(argv)
-
-    save_bash_call(options.outdir)
-
-    phys2denoise(**vars(options))
-
-
-if __name__ == "__main__":
-    _main(sys.argv[1:])
+        LGR.setLevel(logging.INFO)
+
+    physio_file = os.path.abspath(input_file)
+    export_directory = os.path.abspath(export_directory)
+
+    wf = Workflow(
+        name="metrics_wf",
+        input_spec=[
+            "phys",
+            "fs",
+            "mode",
+            "metrics",
+            "metric_args",
+            "metrics_to_export",
+            "bids_parameters",
+            "bids_channel",
+            "tr",
+        ],
+        phys=physio_file,
+        fs=fs,
+        mode=mode,
+        metrics=metrics,
+        metric_args=metric_args,
+        bids_parameters=bids_parameters,
+        bids_channel=bids_channel,
+        metrics_to_export=metrics_to_export,
+        tr=tr,
+    )
+    wf.add(
+        transform_to_physio(
+            name="transform_to_physio",
+            input_file=wf.lzin.phys,
+            fs=wf.lzin.fs,
+            mode=wf.lzin.mode,
+            bids_parameters=wf.lzin.bids_parameters,
+            bids_channel=wf.lzin.bids_channel,
+        )
+    )
+    wf.add(
+        tasks.compute_metrics(
+            name="compute_metrics",
+            phys=wf.transform_to_physio.lzout.out,
+            metrics=wf.lzin.metrics,
+            args=wf.lzin.metric_args,
+        )
+    )
+    wf.add(
+        tasks.export_metrics(
+            name="export_metrics",
+            phys=wf.compute_metrics.lzout.out,
+            metrics=wf.lzin.metrics_to_export,
+            outdir=export_directory,
+            tr=wf.lzin.tr,
+        )
+    )
+    wf.set_output([("result", wf.compute_metrics.lzout.out)])
+
+    return wf
+
+
+def run(workflow: Workflow, plugin="cf", **plugin_args):
+    with Submitter(plugin=plugin, **plugin_args) as sub:
+        sub(workflow)
+
+    return workflow.result()
+
+
+# @due.dcite(
```

Comment:
Apologies if I missed it, but can you make sure that the duecite calls are present everywhere you need them, including the main workflow?

Then we can remove all this commented code wall. It's going to be there in the history of the commits.

Reply:
Yes, my bad, I wanted to relocate the duecite call. I'll fix